app/src/main/java/com/islaharper/dawnofandroid/presentation/screens/login/LoginScreen.kt

```diff
@@ -47,47 +55,123 @@ fun LoginScreen(
                 signedInState = signedInState,
                 messageBarState = messageBarState,
                 onButtonClicked = {
-                    loginViewModel.saveSignedInState(signedIn = true)
-                },
+                    loginViewModel.oneTapSignIn()
+                }
             )
-        },
+        }
+    )
+
+    val launcher = rememberLauncherForActivityResult(
+        ActivityResultContracts.StartIntentSenderForResult()
+    ) { result ->
+        try {
+            if (result.resultCode == Activity.RESULT_OK) {
+                loginViewModel.verifyToken(intent = result.data)
+            } else {
+                loginViewModel.saveSignedInState(signedIn = false)
+            }
+        } catch (e: ApiException) {
+            loginViewModel.saveSignedInState(signedIn = false)
+        }
+    }
+
+    fun launch(signInResult: BeginSignInResult) {
```

Comment:
We can inline this instead of using a local function (Ctl+Alt+N). Local functions make sense where there is recursion, almost where the local function is the heart of some other function.

Reply:
Refactored and committed changes.